packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -1,276 +1,43 @@
-  disableColumnResize: true,
-  disableColumnReorder: true,
-  disableMultipleColumnsFiltering: true,
-  disableMultipleColumnsSorting: true,
-  disableMultipleSelection: true,
-  pagination: true,
-  apiRef: undefined,
-  onRowsScrollEnd: undefined,
-  checkboxSelectionVisibleOnly: false,
-};
-
-export type DataGridProps = Omit<
-  GridComponentProps,
-  | 'apiRef'
-  | 'checkboxSelectionVisibleOnly'
-  | 'disableColumnResize'
-  | 'disableColumnReorder'
-  | 'disableMultipleColumnsFiltering'
-  | 'disableMultipleColumnsSorting'
-  | 'disableMultipleSelection'
-  | 'licenseStatus'
-  | 'options'
-  | 'onRowsScrollEnd'
-  | 'pagination'
-  | 'signature'
-  | 'scrollEndThreshold'
-  | 'selectionModel'
-> & {
-  apiRef?: undefined;
-  checkboxSelectionVisibleOnly?: false;
-  disableColumnResize?: true;
-  disableColumnReorder?: true;
-  disableMultipleColumnsFiltering?: true;
-  disableMultipleColumnsSorting?: true;
-  disableMultipleSelection?: true;
-  onRowsScrollEnd?: undefined;
-  selectionModel?: GridRowId | GridRowId[];
-  pagination?: true;
-};
-
-const MAX_PAGE_SIZE = 100;
+import { DATA_GRID_PROPTYPES } from './DataGridPropTypes';
+import {
+  DEFAULT_GRID_PROPS_FROM_OPTIONS,
+  GridBody,
+  GridErrorHandler,
+  GridFooterPlaceholder,
+  GridHeaderPlaceholder,
+  GridRoot,
+  useGridApiRef,
+} from '../../_modules_/grid';
+import { GridContextProvider } from '../../_modules_/grid/context/GridContextProvider';
+import { useDataGridComponent } from './useDataGridComponent';
+import { DataGridProps } from './DataGridProps';
+import { useDataGridProps } from './useDataGridProps';
 
 const DataGridRaw = React.forwardRef<HTMLDivElement, DataGridProps>(function DataGrid(
   inProps,
   ref,
 ) {
-  const props = useThemeProps({ props: inProps, name: 'MuiDataGrid' });
-  const { pageSize: pageSizeProp, selectionModel: dataGridSelectionModel, ...other } = props;
-
-  let pageSize = pageSizeProp;
-  if (pageSize && pageSize > MAX_PAGE_SIZE) {
-    pageSize = MAX_PAGE_SIZE;
-  }
-
-  const selectionModel =
-    dataGridSelectionModel !== undefined && !Array.isArray(dataGridSelectionModel)
-      ? [dataGridSelectionModel]
-      : dataGridSelectionModel;
+  const props = useDataGridProps(inProps);
+  const apiRef = useGridApiRef();
+  useDataGridComponent(apiRef, props);
 
   return (
-    <GridComponent
-      ref={ref}
-      pageSize={pageSize}
-      {...other}
-      {...FORCED_PROPS}
-      selectionModel={selectionModel}
-      licenseStatus="Valid"
-      signature="DataGrid"
-    />
+    <GridContextProvider apiRef={apiRef} props={props}>
+      <GridRoot ref={ref}>
+        <GridErrorHandler>
+          <GridHeaderPlaceholder />
+          <GridBody />
+          <GridFooterPlaceholder />
+        </GridErrorHandler>
+      </GridRoot>
+    </GridContextProvider>
   );
 });
 
+DataGridRaw.defaultProps = DEFAULT_GRID_PROPS_FROM_OPTIONS;
+
 export const DataGrid = React.memo(DataGridRaw);
 
 // @ts-ignore
-DataGrid.propTypes = {
```

Comment:
I didn't notice this change. I think that we should revert the `propTypes` change. The `propTypes` needs to be in the same file for https://github.com/oliviertassinari/babel-plugin-transform-react-remove-prop-types to trigger. Actually, we also need to replace the TypeScrip bundler to use the Babel bundler of the core that leverages this optimization. Once we move the infra to v5. the prop-types will be automatically generated and in the same file.

Reply:
https://github.com/mui-org/material-ui-x/pull/2176/files/5ca14f2e2feaa13e2ce70bc67130fdbb85d2eb79#r680403715

https://github.com/mui-org/material-ui-x/blob/master/packages/grid/data-grid/src/DataGrid.tsx#L44

I don't know why their is an outdated code visible here :thinking: 
However I forgot to delete the file containing the split prop types, I'll do a PR and merge it as it is trivial

Reply:
#2432